src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Tooltip/MarkupTagHelperTooltipFactory.cs

```diff
@@ -73,14 +73,17 @@ internal static class MarkupTagHelperTooltipFactory
                 descriptionBuilder.Append(finalSummaryContent);
             }
 
-            var availability = await solutionQueryOperations
-                .GetProjectAvailabilityTextAsync(documentFilePath, tagHelperType, cancellationToken)
-                .ConfigureAwait(false);
-
-            if (availability is not null)
+            if (documentFilePath is not null)
             {
-                descriptionBuilder.AppendLine();
-                descriptionBuilder.Append(availability);
+                var availability = await solutionQueryOperations
```

Comment:
❓ reading this I would have expected `ISolutionQueryOperations.GetProjectAvailabilityTextAsync` to handle a null file path and return null. Is there a design decision why it doesn't?

Reply:
Good question! The reason is primarily because there's a cost associated with calling `GetProjectAvailabilityTextAsync(...)`. If the null check happens in `GetProjectAvailabilityTextAsync(...)` we have to make an async call and continue on the thread pool just to perform a null check. That seemed unnecessarily wasteful to me.

Another more opinionated meta reason is that the contract of `GetProjectAvailabilityTextAsync(...)` is to take a document file path and do something with it. By using a non-nullable `string` the API states its intent and makes it clear that it needs a document file path to function. By making it nullable, I think the contract is weaker. That might just be my opinion, but non-nullable reference types are all about stating intent, and I don't think `GetProjectAvailabilityTextAsync(...)` ever _wants_ to be passed a null document file path, so it probably should be clear about that. At least, that was my thought process.